src/lib/mongoBackend/safeMongo.cpp

```diff
@@ -106,6 +106,34 @@ std::string getStringField(const BSONObj& b, const std::string& field, const std
   return "";
 }
 
+
+
+/* ****************************************************************************
+*
+* getNumberField -
+*/
+double getNumberField(const BSONObj& b, const std::string& field, const std::string& caller, int line)
+{
+  if (b.hasField(field) && b.getField(field).type() == NumberDouble)
+  {
+    return b.getField(field).Number();
+  }
+
+  // Detect error
+  if (!b.hasField(field))
+  {
+    LM_E(("Runtime Error (double field '%s' is missing in BSONObj <%s> from caller %s:%d)", field.c_str(), b.toString().c_str(), caller.c_str(), line));
+  }
+  else
+  {
+    LM_E(("Runtime Error (field '%s' was supposed to be an double but type=%d in BSONObj <%s> from caller %s:%d)",
+          field.c_str(), b.getField(field).type(), b.toString().c_str(), caller.c_str(), line));
+  }
+  return -1;
```

Comment:
In general for all the functions in safeMongo, the caller is not able to detect the error. That is intentional: before safeMongo fails of this kind make the CB crash. So the apporach now is to not interrupt the flow (we have to return something... -1 seems to be a good value as use to be very weird to see it, so the user can detect also in the payload) and mark the error in the logs.

However, looking to how getStringFieldF() is implemented, returning "" in this case, maybe we should return 0, which is more neutral.